src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -235,75 +235,82 @@ private async Task FillRequestLists(List<EntryComponent> bundleEntries, Cancella
             _requestCount = bundleEntries.Count;
 
             // For a transaction, we need to resolve any references between resources.
-            // Loop through the entries and if we're POSTing with an ID in the fullUrl then set an ID for it and add it to our dictionary.
+            // Loop through the entries and if we're POSTing with an ID in the fullUrl or doing a conditional create then set an ID for it and add it to our dictionary.
             if (_bundleType == BundleType.Transaction)
             {
                 PopulateReferenceIdDictionary(bundleEntries, _referenceIdDictionary);
             }
 
             foreach (EntryComponent entry in bundleEntries)
             {
-                string persistedId = default;
-
                 if (entry.Request?.Method == null)
                 {
                     _emptyRequestsOrder.Add(order++);
                     continue;
                 }
 
-                HttpContext httpContext = new DefaultHttpContext { RequestServices = _requestServices };
+                await GenerateRequest(entry, order++, cancellationToken);
+            }
+        }
 
-                // For resources within a transaction, we need to resolve any intrabundle references and potentially persist any internally assigned ids
-                if (_bundleType == BundleType.Transaction && entry.Resource != null)
-                {
-                    var requestUrl = (entry.Request != null) ? entry.Request.Url : null;
-                    await _referenceResolver.ResolveReferencesAsync(entry.Resource, _referenceIdDictionary, requestUrl, cancellationToken);
+        private async Task GenerateRequest(EntryComponent entry, int order, CancellationToken cancellationToken)
+        {
+            string persistedId = default;
+            HttpContext httpContext = new DefaultHttpContext { RequestServices = _requestServices };
+
+            var requestUrl = entry.Request?.Url;
+
+            // For resources within a transaction, we need to resolve any intrabundle references and potentially persist any internally assigned ids
+            HTTPVerb requestMethod = entry.Request.Method.Value;
```

Comment:
The assignment for requestUrl has a null check on Request. Is one needed here too?

Reply:
No, where `GenerateRequest` is called above has a null check already for `if (entry.Request?.Method == null)`.